pkg/metricscollector/v1beta1/file-metricscollector/file-metricscollector_test.go

```diff
@@ -19,40 +19,43 @@ package sidecarmetricscollector
 import (
 	"os"
 	"path/filepath"
-	"reflect"
 	"testing"
 	"time"
 
+	"github.com/google/go-cmp/cmp"
 	commonv1beta1 "github.com/kubeflow/katib/pkg/apis/controller/common/v1beta1"
 	v1beta1 "github.com/kubeflow/katib/pkg/apis/manager/v1beta1"
 	"github.com/kubeflow/katib/pkg/controller.v1beta1/consts"
 )
 
-var testJsonDataPath = filepath.Join("testdata", "JSON")
+var (
+	testJsonDataPath = filepath.Join("testdata", "JSON")
+	testTextDataPath = filepath.Join("testdata", "TEXT")
+)
 
 func TestCollectObservationLog(t *testing.T) {
-
-	if err := generateTestFiles(); err != nil {
+	if err := generateJsonTestFiles(); err != nil {
 		t.Fatal(err)
 	}
 	defer os.RemoveAll(filepath.Dir(testJsonDataPath))
 
-	// TODO (tenzen-y): We should add tests for logs in TEXT format.
-	// Ref: https://github.com/kubeflow/katib/issues/1756
-	testCases := []struct {
-		description string
-		filePath    string
-		metrics     []string
-		filters     []string
-		fileFormat  commonv1beta1.FileFormat
-		err         bool
-		expected    *v1beta1.ObservationLog
+	if err := generateTextTestFiles(); err != nil {
+		t.Fatal(err)
+	}
+	defer os.RemoveAll(filepath.Dir(testTextDataPath))
+
+	testCases := map[string]struct {
+		filePath   string
+		metrics    []string
+		filters    []string
+		fileFormat commonv1beta1.FileFormat
+		err        bool
```

Comment:
Exposing error and comparing that error would be better, but I'm ok with working on it as a separate PR.

Example: https://github.com/kubeflow/katib/blob/a2f3fcae55d608a850eaea7ff3d25c667d2423e1/pkg/util/v1beta1/katibconfig/config_test.go#L42-L80

@Electronic-Waste Where do you want to work on it? This PR? Another PR?

Reply:
Another PR is better. I'll open another issue.

Reply:
I'm ok.